### ISSUE TITLE
JupyterHub: how can a user handle limited resources

### ISSUE THREAD

**Issue by martinpovolny** [operate-first/continuous-deployment#93]
If a user of JH requests an environment that does not fit to the cluster, how can he/she cancel the request and/or request another (smaller) environment?

<img width="823" alt="Screenshot 2021-01-15 at 16 15 20" src="https://user-images.githubusercontent.com/51095/104746222-46fdc080-574f-11eb-84e8-c7dd0a703435.png">

I do not see an option in the UI to deal with this situation.

*This is a temporary issue that needs to be created in the right repository/project or linked from such a place.

**Comment by vpavlin** [comment]
Yeah, this is a problem - we kept bumping the timeout because of the size of some of the images.

I am not sure if there is an option to cancel schedulling in the JH code, we need to check that.

We also need to start collecting these errors somewhere and provide them to the admins of the ODH deployment

**Comment by tumido** [comment]
> We also need to start collecting these errors somewhere and provide them to the admins of the ODH deployment

@vpavlin for ODH SRE a simple metrics for scheduling duration would be enough I think, you can alert on that. Also these events are kubernetes events, right. So it can be scraped for SRE as well.

cc @hemajv WDYT?

**Comment by vpavlin** [comment]
Yeah, I agree anomalies in start times would be great alert for us and for admins. If we could map it to events, that would be even more awesome:-)

**Comment by anishasthana** [comment]
I believe Jupyterhub already has _some_ metrics around start time. https://jupyterhub.readthedocs.io/en/latest/reference/metrics.html 

Do `jupyterhub_server_spawn_duration_seconds` and 
`jupyterhub_init_spawners_duration_seconds ` work for this?

**Comment by hemajv** [comment]
Yeah I think its a good idea to have some alerting on these metrics. As @anishasthana pointed out, there are some metrics around start time for spawners/servers/hub that we can look into.
